Amazon.IonObjectMapper.Test/IonObjectSerializerTest.cs

```diff
@@ -168,6 +168,29 @@ public void ExceptionOnMultiParameterIonPropertySetterMethods()
             Assert.ThrowsException<InvalidOperationException>(() => serializer.Deserialize<Chalkboard>(stream));
         }
 
+        [TestMethod]
+        public void DeserializesObjectsWithIonConstructor()
+        {
+            var wheel = new Wheel("default", 17, 20, "MSW", "black");
+
+            var stream = defaultSerializer.Serialize(wheel);
+            var deserialized = defaultSerializer.Deserialize<Wheel>(stream);
+
+            // Verify custom IonConstructor was used and remaining properties/fields were set after construction.
+            Assert.AreEqual(
+                $"Specification: {wheel.Specification}, Size: {wheel.Size} inches", deserialized.Specification);
+            Assert.AreEqual(wheel.offset, deserialized.offset);
+            Assert.AreEqual(wheel.Brand, deserialized.Brand);
+            Assert.AreEqual(wheel.color, deserialized.color);
+        }
```

Comment:
`Check` does not work here as the deserialized `Specification` property differs from the original (which is how we verify that the custom constructor was used).

I tried altering the `Specification` to be a field (which is ignored by default) and then let the custom constructor set the field but that does not work because the field is ignored during serialization and then absent entirely during deserialization (when the custom construction happens).

Reply:
You don't need to "prove the constructor was called" by having code in the constructor. You just need to have an object that can only be created via its constructor, and which set's private fields, and then . Then you can use `Check` as Homayoun suggestst.

Reply:
Now using `Check`.
(we are relying on the expectation that we would get a `No parameterless constructor defined for type 'Amazon.IonObjectMapper.Test.Wheel'` exception if this test invoked `Activator.CreateInstance` rather than the annotated constructor).